### ISSUE TITLE
TypeError: performance.getEntriesByName(...)[0] is undefined in speedtest.js (Zeile 840, Spalte 49)

### ISSUE THREAD

**Issue by kritter** [nesk/network.js#32]
In Firefox, hundreds of this error are thrown because of flawed latency measures. No measures at all are generated due to a temporary server misconfiguration, but Chrome handles this more gracefully than Firefox at the moment.

In Firefox, the cause of this error is caused by using [0] on an undefined.

**Comment by kritter** [comment]
Update: all type of measurements are somehow affected, yielding erroneous results for download and upload speed. Chrome works completely fine.

**Comment by paulovpereira** [comment]
Same problem here. It only happens in Firefox. 

Actually the problem happens because it tries to get a perfomance entry by name but it isn't finding it on the list of entries. Then it tries to get a 'duration' property from position zero of this empty array resulting in error.

```javascript
measures[measureLabel] = performance.getEntriesByName(measureLabel)[0].duration;
```

We noticed that it happens after it tries do upload/download a file bigger than ~1 Mb. After the first error all the following measures fails.

@nesk, any idea?

**Comment by nesk** [comment]
Sorry for taking so much time to respond @kritter.

Well, I could have an idea explaining why this happens but I will need to make some tests to be sure (I will try to do this this week-end). The Timing class is probably at fault.

**Comment by paulovpereira** [comment]
@nesk thank you very much for the fast reply.

If you need any help to do your tests let me know.

**Comment by nesk** [comment]
Sooo! Just got able to dig a bit into this issue. It seems like Firefox has an internal constant `PERFORMANCE_MAX_ENTRY_SIZE` defining how much entries the Performance API can store, this is some bad news because it is currently set to `150`… This is based on [this comment](https://bugzilla.mozilla.org/show_bug.cgi?id=822480#c141), however I can't find the specified constant in Firefox's sources, it may have been renamed or whatever but I can clearly see that Firefox stops storing new Performance entries once the 150 cap is reached.

The good news is: I _think_ I can fallback on a `Date` solution when the maximum is reached. I will try work on this in the next days.

**Comment by nesk** [comment]
FYI, it seems like the `PERFORMANCE_MAX_ENTRY_SIZE ` as been renamed `kDefaultBufferSize` and is defined in __dom/base/nsPerformance.h__ L372 (in the v39 sources).

**Comment by nesk** [comment]
This bug should be fixed, you can try this with the v2.0.0-alpha4 release.

**Comment by paulovpereira** [comment]
We tested here with v2.0.0-alpha4 and it's working just fine in Firefox. Thank you very much!
